File: frontend/src/api/analysis.ts
import { api } from "./client";

export async function getBottlenecks() {
  const response = await api.get("/analyze/bottlenecks");
  return response.data.items ?? [];
}

export async function getCriticalPath(milestoneId: string) {
  const response = await api.get(`/analyze/critical-path/${milestoneId}`);
  return response.data;
}

export async function getTeamWorkload() {
  const response = await api.get("/analyze/workload");
  return response.data.items ?? [];
}

export async function getStaleBlockers(thresholdDays = 14) {
  const response = await api.get("/analyze/stale-blockers", { params: { threshold_days: thresholdDays } });
  return response.data.items ?? [];
}

export async function getHandoffHotspots() {
  const response = await api.get("/analyze/handoff-hotspots");
  return response.data.items ?? [];
}

export async function runImpactAnalysis(itemId: string, slipDays: number) {
  const response = await api.post("/analyze/impact", { item_id: itemId, slip_days: slipDays });
  return response.data;
}

File: src/lance_graph_project/api/routers/analysis.py
from __future__ import annotations

from fastapi import APIRouter

from lance_graph_project.api.schemas import ImpactRequest
from lance_graph_project.services.container import get_analysis_service

router = APIRouter(prefix="/api/analyze", tags=["analysis"])


@router.get("/bottlenecks")
def get_bottlenecks(workstream: str | None = None, top_k: int = 10) -> dict:
    items = get_analysis_service().get_bottlenecks(workstream=workstream, top_k=top_k)
    return {"items": items, "count": len(items)}


@router.get("/critical-path/{milestone_id}")
def get_critical_path(milestone_id: str) -> dict:
    return get_analysis_service().get_critical_path(milestone_id=milestone_id)


@router.post("/impact")
def run_impact_analysis(request: ImpactRequest) -> dict:
    return get_analysis_service().run_impact_analysis(item_id=request.item_id, slip_days=request.slip_days)


@router.get("/workload")
def get_workload(team: str | None = None) -> dict:
    items = get_analysis_service().get_team_workload(team_id=team)
    return {"items": items, "count": len(items)}


@router.get("/handoff-hotspots")
def get_handoff_hotspots() -> dict:
    items = get_analysis_service().get_handoff_hotspots()
    return {"items": items, "count": len(items)}


@router.get("/stale-blockers")
def get_stale_blockers(threshold_days: int = 14) -> dict:
    items = get_analysis_service().get_stale_blockers(threshold_days=threshold_days)
    return {"items": items, "count": len(items)}
